### PR TITLE
sendReceiveConfirm on invite

### PR DESCRIPTION
**Pull request by embbnux**
to fix problem that `toVoicemail` and `sendToMessage` will fail sometimes.

### CODE REVIEW

File: src/ringcentral-web-phone.js [hunk 1]
```diff
@@ -206,6 +206,11 @@
             this.userAgent.audioHelper.playIncoming(true);
             patchSession(session);
             patchIncomingSession(session);
+            session.sendReceiveConfirm().then(function() {
+                session.logger.log('sendReceiveConfirm success');
+            }).catch(function(error){
+                console.error('failed to send receive confirmation via SIP MESSAGE due to ' + error);
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: Console output is forbidden in library code. You may use session logger if needed, but you have to also emit an event. Please talk to @vyshakhbabji as he did some research about it.

Comment (embbnux) [reply src/ringcentral-web-phone.js]: OK, I will update it.

**Comment by kirill-konshin** [comment]
@vyshakhbabji please verify

File: src/ringcentral-web-phone.js [hunk 2]
```diff
@@ -206,6 +206,12 @@
             this.userAgent.audioHelper.playIncoming(true);
             patchSession(session);
             patchIncomingSession(session);
+            session._sendReceiveConfirmPromise = session.sendReceiveConfirm().then(function() {
+                session.logger.log('sendReceiveConfirm success');
+            }).catch(function(error){
+                session.logger.error('failed to send receive confirmation via SIP MESSAGE due to ' + error);
+                return Promise.reject(error);
```
Comment (kirill-konshin) [inline src/ringcentral-web-phone.js]: Throw the error again:

```js
throw error;
```

Comment (embbnux) [reply src/ringcentral-web-phone.js]: `throw error` is equal as `return Promise.reject(error)`.   If you prefer it, I will update.

Comment (kirill-konshin) [reply src/ringcentral-web-phone.js]: Yes, rethrow is more appropriate

**Pull request merged**